core/templates/dev/head/domain/topic/SubtopicObjectFactory.js

```diff
@@ -17,12 +17,16 @@
  * subtopic domain objects.
  */
 
-oppia.factory('SubtopicObjectFactory', [
-  function() {
-    var Subtopic = function(subtopicBackendObject) {
+oppia.factory('SubtopicObjectFactory', ['SkillSummaryObjectFactory',
+  function(SkillSummaryObjectFactory) {
+    var Subtopic = function(subtopicBackendObject, skillIdToDescriptionMap) {
```

Comment:
The Subtopic constructor should take individual fields, not an entire backend object. (The constructor for a domain object in the frontend should not even know about backend objects.)

Reply:
Done.